### ISSUE TITLE
`No such symbol` error on installing package that uses rstantools

### ISSUE THREAD

**Issue by mikekaminsky** [stan-dev/rstantools#72]
Hi 👋 . I feel like I'm on the _cusp_ of getting `rstantools` to work for my package.  Apologies if this is a dumb question or if I'm missing something obvious.

I'm getting an error at the very end of package installation and I can't for the life of me trace what the issue might be. Here's the error:

```
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘recast’ in .doLoadActions(where, attach):
 error in load action .__A__.1 for package recast: Rcpp::loadModule(module = "stan_fit4state_space_with_kf_mod", : Unable to load module "stan_fit4state_space_with_kf_mod": Failed to initialize module pointer: Error in FUN(X[[i]], ...): no such symbol _rcpp_module_boot_stan_fit4state_space_with_kf_mod in package recast
```

The issue seems to be related to how we register the modules with `Rcpp` [here](https://github.com/stan-dev/rstantools/blob/8dc2f8a982314859066690d12f79ef43dd3a368a/R/rstan_config.R#L217). Of course `state_space_with_kf` is the name of our stan model. 

However, I confirmed that `rstantools::rstan_config()` completes successfully in the package, so I'm not sure what to check next. Any ideas would be super-appreciated.

Environment Notes:
* Running on macOS
* R 3.6
* I used `rstantools::use_rstan()` to get setup

**Comment by mikekaminsky** [comment]
I forgot to check `src/`  into source control. I suspect that's the problem. Apologies!

**Comment by mikekaminsky** [comment]
Okay! I think I fixed this. Going to describe what I had to do to fix it for posterity / future folks fixing this problem. Happy to close the issue or leave it open depending on what the maintainers want.

The big issue is that I was confused about the directions to change the `NAMESPACE` file. At the top of the `NAMESPACE` file it says, very clearly

```
# Generated by roxygen2: do not edit by hand
```

That of course conflicts with the directions from `rstantools` that you need to add lines to the namespace file. I couldn't figure out how `rstantools` and `roxygen2` were supposed to play together. Through a fair amount of debugging, I figured out that adding the `<package-name>-package.R` file that gets created when you run `rstan::rstan_create_package()` allows roxygen to manage the files correctly. (At least, I think that's what made the difference).

I think this could be more clear in the docs / vignette what you're supposed to do with a package that's already using roxygen -- happy to take a stab at making those updates if someone can confirm what the expected behavior here is (even if it's "Ignore the line at the top of NAMESPACE file").

**Comment by jgabry** [comment]
Thanks @mikekaminsky and sorry for the hassle/confusion! We'd happily accept a PR to make this more clear if you want to take a stab at it. I think what you ended up figuring out is indeed what was intended. (is that right @mlysy?)

**Comment by mikekaminsky** [comment]
Happy to take a stab at that! 

Additionally, maybe the feature to add here is to include the creation of `<package-name>-package.R` file if it doesn't already exist when calling `rstantools::use_rstan()`?

I'm sort of a `roxygen` noob so maybe everyone else already has that file setup and generally won't experience this issue. Will wait for confirmation on the above before opening a PR to update the docs :)

**Comment by jgabry** [comment]
> Additionally, maybe the feature to add here is to include the creation of `<package-name>-package.R` file if it doesn't already exist when calling `rstantools::use_rstan()`?

Yeah good point!

**Comment by mlysy** [comment]
My understanding is that `<package-name>-package.R` is the standard roxygen way of editing the `NAMESPACE` file.  I agree that the error message above is completely unintelligible unless you already know what it means.  On the other hand, since rstantools does not force users to depend on roxygen, I don't we should be responsible for errors due to being unfamiliar with how it works.

I suppose `use_rstan()` could add `<package-name>-package.R` if it doesn't already exist with the correct dependencies.  But adding roxygen tags to such a file if it already exists involves extensive parsing which seems quite complicated (even [usethis](https://usethis.r-lib.org/) won't do this for you unless it created the `*-package.R` file itself).  

A simpler alternative might be to modify `use_rstan()`'s instructions regarding the `NAMESPACE` to e.g.,
```
Add the following lines (e.g., using roxygen) to the NAMESPACE:
```
?

**Comment by mikekaminsky** [comment]
Got it!

I'd suggest adding `<package-name>-package.R` if it doesn't already exist. Since you do that for new packages anyway that doesn't seem like a _huge_ stretch. I would definitely not muck with an already existing file and instead just warning like `Identified <package-name>-package.R. Skipping template creation` or similar.

FWIW, I don't think I ever formally "set up" Roxygen but rather it was installed by some other package (maybe I did it originally via `use_this`? who knows). I know I've never edited the namespace file. I could see a user who sets up their package with `rstantools` being confused down the road about the purpose of the `<package-name>-package.R` file if they didn't take the time to go read the `roxygen` docs. 

I think the improved instructions / docs alone could go along away. I'd go with 

```
Add the following lines (e.g., via <package-name>-package.R if using roxygen) to the NAMESPACE:
```

with maybe something slightly more spelled-out in the vignette. 

I've got some time carved out this weekend to take a stab at updating the docs at least.
